test/Microsoft.AspNet.Routing.Tests/RouteCollectionTest.cs

```diff
@@ -17,6 +18,35 @@ namespace Microsoft.AspNet.Routing
 {
     public class RouteCollectionTest
     {
+        [Theory]
+        [InlineData(@"Home/Index/23", "home/index/23", true)]
+        [InlineData(@"Home/Index/23", "Home/Index/23", false)]
+        [InlineData(@"Home/Index/23?Param1=ABC&Param2=Xyz", "Home/Index/23?Param1=ABC&Param2=Xyz", false)]
+        [InlineData(@"Home/Index/23?Param1=ABC&Param2=Xyz", "home/index/23?Param1=ABC&Param2=Xyz", true)]
+        [InlineData(@"Home/Index/23#Param1=ABC&Param2=Xyz", "Home/Index/23#Param1=ABC&Param2=Xyz", false)]
+        [InlineData(@"Home/Index/23#Param1=ABC&Param2=Xyz", "home/index/23#Param1=ABC&Param2=Xyz", true)]
+        public void GetVirtualPath_CanLowerCaseUrls_BasedOnOptions(
```

Comment:
Please include a case with turkish i. We're depending on invariant culture in this test so let's verify that fact.